router/core/header_rule_engine.go

```diff
@@ -76,6 +77,19 @@ func (h HeaderRuleEngine) OnOriginRequest(request *http.Request, ctx RequestCont
 	for _, rule := range requestRules {
 		// Forwards the matching client request header to the upstream
 		if rule.Operation == config.HeaderRuleOperationPropagate {
+			// Rename the header
+			if rule.Rename != "" {
```

Comment:
This means rename would only work with an exact match but we also provide a regex match.

Reply:
done. Added the logic in matching block.